src/ol/source/imagemapguidesource.js

```diff
@@ -150,6 +153,8 @@ ol.source.ImageMapGuide.prototype.getImage =
     image = new ol.Image(extent, resolution, pixelRatio,
         this.getAttributions(), imageUrl, this.crossOrigin_,
         this.imageLoadFunction_);
+    goog.events.listen(image, goog.events.EventType.CHANGE,
+        this.handleImageChange, false, this);
```

Comment:
in ```disposeInternal``` of ```ol.Image``` similar approach used as here: https://github.com/openlayers/ol3/pull/3248